src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -600,6 +611,10 @@ public PossiblyConditionalState Clone()
 
             void learnFromNonNullTest(int inputSlot, ref LocalState state)
             {
+                if (stateWhenNotNullOpt is { } stateWhenNotNull && inputSlot == originalInputSlot)
```

Comment:
>stateWhenNotNull [](http://example.com/codeflow?start=47&length=16)

nit: Here too, it feels like we shouldn't need the extra local, even to satisfy nullability checks on `CloneAndUnsplit(ref ...)`

Reply:
Problem here is we can't take a `ref` to `stateWhenNotNullOpt.Value`.